backend/adapter_processor/views.py

```diff
@@ -229,7 +230,17 @@ def destroy(
             logger.error("Cannot delete a default adapter")
             raise CannotDeleteDefaultAdapter()
 
-        super().perform_destroy(adapter_instance)
+        try:
+            super().perform_destroy(adapter_instance)
+        except ProtectedError:
```

Comment:
Do we get the dependent object from this error? If  available do you think displaying the name and type in error message would be a good idea?

Reply:
@johnyrahul we do get it from the exception. However I haven't checked how it appears in case of multiple usages. Also I'm not sure if returning all the dependencies in the same API is a good practice or if we need a separate dependency check API returning this.
Either way, since our new model to track usages would solve this, I just marked it as a TODO

Reply:
Also we need to do this generically for other places where we check for dependencies

Reply:
@tahierhussain @mohamed-siddhiq @vishnuszipstack your thoughts on this? 
1. Returning dependency related information as a separate API -> allows us to display anything in the FE to show these and even allow some actions like deleting / removing usage from the same screen
2. Returning dependency related information in the same delete API -> It would always be rendered as a toast message I believe, can we render the information differently in this case?

Personally I think we should have either (should be a product call though)
a. actions supporting easy delete of the dependent workflows / prompt studio projects
b. links to these workflows / prompt studio projects to quickly go to them

Reply:
Closing this conversation to unblock this PR. Suggested change can be taken up as a separate PR